pkg/bgpv1/manager/workdiff.go

```diff
@@ -16,31 +16,31 @@ import (
 type reconcileDiff struct {
 	// incoming CiliumBGPVirtualRouter configs mapped by their
 	// local ASN.
-	seen map[int]*v2alpha1api.CiliumBGPVirtualRouter
+	seen map[int64]*v2alpha1api.CiliumBGPVirtualRouter
 	// the state of the bgp control plane at the time of this reconcileDiff's
 	// creation.
 	state *agent.ControlPlaneState
 	// Local ASNs which BgpServers must be instantiated, configured,
 	// and added to the manager. Intended key for `seen` map.
-	register []int
+	register []int64
```

Comment:
From the k8s api conventions:
> All public integer fields MUST use the Go (u)int32 or Go (u)int64 types, not (u)int (which is ambiguous depending on target platform). Internal types may use (u)int.

Since this is an internal type, should we continue using `int`?

Reply:
I don't think so, it could cause issues on 32-bit systems - think of storing max 32-bit ASN (`4294967295`) in 32-bit int.